fieldpath/set_test.go

```diff
@@ -664,80 +663,227 @@ func TestSetDifference(t *testing.T) {
 	}
 }
 
-var nestedSchema = func() (*schema.Schema, schema.TypeRef) {
-	sc := &schema.Schema{}
+var nestedSchema = func() (*typed.Parser, string) {
 	name := "type"
-	err := yaml.Unmarshal([]byte(`types:
+	parser := mustParse(`types:
 - name: type
   map:
     elementType:
       namedType: type
     fields:
+      - name: keyAStr
+        type:
+          scalar: string
+      - name: keyBInt
+        type:
+          scalar: numeric
       - name: named
         type:
           namedType: type
       - name: list
         type:
           list:
-            elementRelationShip: associative
-            keys: ["name"]
+            elementRelationship: associative
```

Comment:
how much of the test changes came from fixing this typo and how many were optional additions / modifications made? ideally, we'd fix bugs in the test separately, and avoid optional modifications concurrent with changing the behavior being tested